pacsman/pynetdicom_client.py

```diff
@@ -0,0 +1,399 @@
+                                    image_ids.append(instance.SOPInstanceUID)
+                            else:
+                                raise Exception(
+                                    'Image C-FIND Failure Response: 0x{0:04x}'.format(
+                                        status.Status))
+
+                    info = SeriesInfo(series_id=series.SeriesInstanceUID, description=description,
+                                      modality=series.Modality, num_images=len(image_ids),
+                                      acquisition_datetime=series.SeriesDate)
+
+                    series_infos.append(info)
+
+        return series_infos
+
+    def fetch_images_as_files(self, series_id):
+        """
+        Fetches series images from PACS with C-MOVE/C-STORE
+        :param series_id: SeriesInstanceUID from PACS
+        :return: a path to a directory full of dicom files on success, None if not found
+        """
+        series_path = os.path.join(self.dicom_dir, series_id)
+        scp = StorageSCP(self.client_ae, series_path)
+        scp.start()
+
+        try:
+            ae = AE(ae_title=self.client_ae,
+                    scu_sop_class=QueryRetrieveSOPClassList,
+                    transfer_syntax=[ExplicitVRLittleEndian])
+
+            ext_neg = []
+            for context in ae.presentation_contexts_scu:
+                tmp = SCP_SCU_RoleSelectionNegotiation()
+                tmp.sop_class_uid = context.abstract_syntax
+                tmp.scu_role = False
+                tmp.scp_role = True
+                ext_neg.append(tmp)
+
+            with association(ae, self.pacs_url, self.pacs_port, ext_neg=ext_neg) as assoc:
+                dataset = Dataset()
+                dataset.SeriesInstanceUID = series_id
+                dataset.QueryRetrieveLevel = 'IMAGE'
+
+                if scp.is_alive():
+                    responses = assoc.send_c_move(dataset, scp.ae_title,
+                                              query_model='S')
+                else:
+                    raise Exception(f'Storage SCP failed to start for series {series_id}')
+
+                for (status, response) in responses:
+                    logger.debug(status)
+                    logger.debug(response)
+
+                    if status.Status not in status_success_or_pending:
+                        raise Exception(
+                            'Image C-MOVE Failure Response: 0x{0:04x}'.format(
+                                status.Status))
+
+                return series_path if os.path.exists(series_path) else None
+
+        except Exception as e:
+            raise e
+        finally:
+            scp.stop()
+
+    def fetch_thumbnail(self, series_id):
+        """
+        Fetches central slice of a series from PACS with C-GET
+        :param series_id: SeriesInstanceUID from PACS
+        :return: A path to a dicom file on success, None if not found
```

Comment:
@yujanshrestha what if we returned the numpy array for the thumbnail?  Returning the path to a DICOM file seems weird to me.

Reply:
I like this idea, but we should chat about the "numpifier" part of pacsman a bit. Even the non-thumbnail `fetch_images` could potentially return numpy arrays.